ledger/src/blockstore.rs

```diff
@@ -179,6 +199,7 @@ pub struct Blockstore {
     no_compaction: bool,
     slots_stats: Arc<Mutex<SlotsStats>>,
     advanced_options: BlockstoreAdvancedOptions,
+    turbine_slots_stats: Arc<Mutex<HashMap<Slot, TurbineSlotStats>>>,
```

Comment:
also, don't need `Arc`.

Reply:
There are 2 separate issues here:
1. The memory usage of this new hashmap will keep growing if some slots linger on. `prune` does help in here but only as long as the validator keeps making roots.
However, when the validator stops making roots it exacerbates this issue: https://github.com/solana-labs/solana/issues/23061
Might need to change outer haspmap type to `LurCache` to avoid the worst case scenario.

2. If we are silently ignoring some of the slots and not emitting final stats for them, then the stats we get at the end are not representative of all slots broadcasted through turbine, and so are not reliable or at least complete.
So we need to make sure at the end the stats for any slots received through turbine are submitted and indicated if the slot finally got rooted, mark dead, etc....

Reply:
1. agreed. Thoughts on making this LRU cache or purging based on time like `SlotsStats` https://github.com/solana-labs/solana/blob/49228573f4af4719c7c56b8fcc89e7fbc11a44a5/ledger/src/blockstore.rs#L1989
2. are you looking for every entry removed from the cache to be logged in the same table with some boolean values to indicate why it was removed? or separate tables?